cli/alert.go

```diff
@@ -0,0 +1,180 @@
+  bash variable expansions. The non-option section of arguments constructs a list
+  of "Matcher Groups" that will be used to filter your query. The following
+  examples will attempt to show this behaviour in action:
+
+  amtool alert query alertname=foo node=bar
+
+  	This query will match all alerts with the alertname=foo and node=bar label
+  	value pairs set.
+
+  amtool alert query foo node=bar
+
+  	If alertname is ommited and the first argument does not contain a '=' or a
+  	'=~' then it will be assumed to be the value of the alertname pair.
+
+  amtool alert query 'alertname=~foo.*'
+
+  	As well as direct equality, regex matching is also supported. The '=~' syntax
+  	(similar to prometheus) is used to represent a regex match. Regex matching
+  	can be used in combination with a direct match.
+
+  amtool alert query alertname=foo node={bar,baz}
+
+  	This query will match all alerts with the alertname=foo label value pair
+  	and EITHER node=bar or node=baz.
+
+  amtool alert query alertname=foo{a,b} node={bar,baz}
+
+	Similar to the previous example this query will match all alerts with any
+	combination of alertname=fooa or alertname=foob AND node=bar or node=baz.
+
+	`,
+	RunE: queryAlerts,
+}
+
+func init() {
+	RootCmd.AddCommand(alertCmd)
+	alertCmd.Flags().Bool("expired", false, "Show expired alerts as well as active")
+	alertCmd.Flags().BoolP("silenced", "s", false, "Show silenced alerts")
+	alertFlags = alertCmd.Flags()
+}
+
+func fetchAlerts() (model.Alerts, error) {
+	alertResponse := alertmanagerAlertResponse{}
+
+	u, err := GetAlertmanagerURL()
+	if err != nil {
+		return model.Alerts{}, err
+	}
+
+	u.Path = path.Join(u.Path, "/api/v1/alerts")
+	res, err := http.Get(u.String())
+	if err != nil {
+		return model.Alerts{}, err
+	}
+
+	defer res.Body.Close()
+	decoder := json.NewDecoder(res.Body)
+
+	err = decoder.Decode(&alertResponse)
+	if err != nil {
+		return model.Alerts{}, errors.New("Unable to decode json response")
+	}
+	return alertResponse.Data, nil
+}
+
+func queryAlerts(cmd *cobra.Command, args []string) error {
+	alerts, err := fetchAlerts()
+	if err != nil {
+		return err
+	}
+
+	silences, err := fetchSilences()
+	if err != nil {
+		return err
+	}
+
+	var groups []types.Matchers
+	if len(args) > 0 {
+		matchers, err := parseMatchers(args, RESOLVE_FUZZY)
+		if err != nil {
+			return err
+		}
+		groups = parseMatcherGroups(matchers)
+		if err != nil {
+			return err
+		}
+	}
+
+	expired, err := alertFlags.GetBool("expired")
+	if err != nil {
+		return err
+	}
+
+	showSilenced, err := alertFlags.GetBool("silenced")
+	if err != nil {
+		return err
+	}
+
+	displayAlerts := model.Alerts{}
+	for _, alert := range alerts {
+		// If we are only returning current alerts and this one has already expired skip it
+		if !expired {
+			if !alert.EndsAt.IsZero() && alert.EndsAt.Before(time.Now()) {
+				continue
+			}
+		}
+
+		if !showSilenced {
```

Comment:
This filtering is something I've also implemented in the new alertmanager-ui. Seems like we should add it as a query string param to the url and do the filtering on the initial query, instead of duplicating our work in different consumers of the api?